bpfd-operator/controllers/bpfd-operator/kprobe-program.go

```diff
@@ -112,6 +112,20 @@ func (r *KprobeProgramReconciler) updateStatus(ctx context.Context, name string,
 		r.Logger.V(1).Info("failed to get fresh KprobeProgram object...requeuing")
 		return ctrl.Result{Requeue: true, RequeueAfter: retryDurationOperator}, nil
 	}
+
+	if prog.Status.Conditions != nil {
+		// Get most recent condition
+		recentIdx := len(prog.Status.Conditions) - 1
+
+		// If the most recent condition is the same as input, just return.
+		if prog.Status.Conditions[recentIdx].Type == string(cond) {
+			return ctrl.Result{}, nil
+		} else {
+			// Remove the input condition from the list if it exists (may no exist)
+			// because the SetStatusCondition() doesn't append if it is already in the list.
+			meta.RemoveStatusCondition(&prog.Status.Conditions, string(cond))
```

Comment:
Urg ok I see https://github.com/kubernetes/apimachinery/blob/v0.27.4/pkg/api/meta/conditions.go#L30

That's really ugly....Are the conditions not reordered when applied though? 

i.e if the conditions were 

1. Loaded
2. NotSelected

And we tried to update back to Loaded......would it just update 1 and move it to the end of the array? or simply update 1 in place?

Reply:
So, if the status value changes, then 1 is updated (not moved to the end). If status is not updated, it's just a noop.

To address, I removed it, then re-add it so it is added to the end of the list.

Reply:
Agree, but didn't think this PR was the right place.